### ISSUE TITLE
issues with docs of scalars in 0.45.0

### ISSUE THREAD

**Issue by NielsCo** [microsoft/typespec#2021]
Hey,
I'm currently working on a PostgreSQL emitter for TypeSpec

When updating my own emitter from 0.44.0 to 0.45.0 I got some weird behavior for docs that I didn't expect.
I have a method that previously applied the docs of custom scalar types to all properties that use it (in the resulting postgreSQL)

So code like this:
```
 @doc("hello")
  scalar MyScalar extends string
  model ScalarId {
      @key id: MyScalar
  }
```
Has the following PostgreSQL output:
`CREATE TABLE ScalarId (/* hello */ id TEXT PRIMARY KEY);`

Now I have a method that implemented this behavior that creates these comments and adds them to the outputed type:

```
 applyDocs(type: Type, target: Documented) {
    // apply docs if we have a custom-scalar
    if (type.kind === "ModelProperty" && type.type.kind === "Scalar") {
      const scalar = type.type;
      this.applyDocs(scalar, target);
    }

    const docs = getDoc(this.program, type);
    if (docs) {
      target.docs = docs;
    }
  }
```

That I had to rewrite to:
```
  applyDocs(type: Type, target: Documented) {
    // apply docs if we have a custom-scalar
    if (type.kind === "ModelProperty" && type.type.kind === "Scalar") {
      const scalar = type.type;
      if (!this.program.checker.isStdType(scalar)) { // check this isn't an included scalar to not get the docs of the scalar
        this.applyDocs(scalar, target);
      }
    }
    const docs = getDoc(this.program, type);
    if (docs) {
      target.docs = docs;
    }
```

Otherwise I would get the comment of the scalar which is in the TypeSpec definition itself.
So for me the output included the comment "`/* A sequence of textual characters. */`"
I also tried to use the getDocData and checked that the doc was only of the type comment. But that didn't work for me.

I sadly don't have time to investigate whether this is intended behavior myself.
Maybe this is just a weird behavior because I have to "flatten" the scalar type in my emitter and still want to keep all the comments of custom scalars

**Comment by timotheeguerin** [comment]
Just to be sure you are saying that you are seeing the doc from TypeSpec std types being included down the line because you flatten your scalars. 
If so this was expected and the openapi3 emitter had to do a similar change. The origin of hte change is that doc comment (`/** */`) can now be used as a way to replace @doc but this meant the doc comment for built-in scalars were included too. 
We did think it was correct to include those docs as if you needed to generate a type for that  built-in scalar you would have a doc for it but as you found out if you flatten the docs you would have to add this extra check.

If this is something that really turns out to be bring more waste than value we can consider not having the doc carry over for those types. Potentially pending on this issue #2003

**Comment by timotheeguerin** [comment]
using the `isStdType` is going to be your best bet right now. Is that causing you some issue? And yeah you shouldn't rely on `getDocData` as you ideally would still let your user set their doc using doc comment.

**Comment by NielsCo** [comment]
Using `isStdType` is working for me.
Just wanted to report this as for me this behavior seemed strange/unexpected. If it's intended you can close this issue

**Comment by timotheeguerin** [comment]
It was expected I'll leave open for now and see if there is more feedback on this and we should do something about it.

**Comment by markcowl** [comment]
Closing in favor of: https://github.com/microsoft/typespec/issues/2003
